src/lib.rs

```diff
@@ -214,6 +214,9 @@ pub use word_splitters::WordSplitter;
 pub mod wrap_algorithms;
 pub use wrap_algorithms::WrapAlgorithm;
 
+pub mod line_ending;
```

Comment:
I think the module is small enough that we can just inline it into `lib.rs`. Or perhaps better: leave it as an internal module and then `pub use` the enum like you already do. Basically, since I don't think the module will contain anything which isn't also exported at the root level, I think we should not expose the module.

Reply:
My recent change adds custom `NonEmptyLines` iterator to that module, which returns the line together with the detected line ending.

Do you still want this code to reside in lib.rs or should we keep it in a dedicated file?

Reply:
Good point, I think you're right that we should keep it as a dedicated file. I don't think there is anything in the module which should be exported, so we can remove the `pub` here.

Reply:
`LineEnding` enum should be exported I think (we allow clients to set it in `Options`)

Reply:
Yep, it should be exported, but this can be done via the `pub use` in `lib.rs`. If you do that and you don't expose the `line_ending` module itself, the enum behaves exactly as if it had been defined in `lib.rs`. You can see that with [`WordSeparator`](https://docs.rs/textwrap/latest/textwrap/enum.WordSeparator.html) which shows up in the crate root, but is actually defined in a private module.